src/XrmMockupShared/Requests/QualifyLeadRequestHandler.cs

```diff
@@ -0,0 +1,153 @@
+                throw new FaultException($"{status} is not a valid status code on lead with Id {request.LeadId.Id}");
+            }
+
+            if (statusOptionMetadata.State != 1)
+            {
+                throw new FaultException($"{request.Status.Value} is not a valid status code for state code LeadState.Qualified lead with Id {request.LeadId.Id}");
+            }
+
+            var createdEntities = new EntityReferenceCollection();
+
+            if (request.CreateAccount)
+            {
+                var accountAlreadyExists = db["account"].Any(row => row["name"] != null && row["name"].Equals(lead["fullname"]));
+
+                if (accountAlreadyExists)
+                {
+                    throw new FaultException("A record was not created or updated because a duplicate of the current record already exists.");
+                }
+
+                var account = new Entity
+                {
+                    Id = Guid.NewGuid(),
+                    LogicalName = "account",
+                };
+
+                createdEntities.Add(account.ToEntityReference());
+            }
+
+            if (request.CreateContact)
+            {
+                var contactAlreadyExists = db["contact"].Any(row => row["fullname"] != null && row["fullname"].Equals(lead["fullname"]));
+
+                if (contactAlreadyExists)
+                {
+                    throw new FaultException("A record was not created or updated because a duplicate of the current record already exists.");
+                }
+
+                var contact = new Entity
+                {
+                    Id = Guid.NewGuid(),
+                    LogicalName = "contact",
+                };
+
+                createdEntities.Add(contact.ToEntityReference());
+            }
+
+            if (request.CreateOpportunity)
+            {
+                if (request.OpportunityCustomerId != null)
+                {
+                    if (request.OpportunityCustomerId.LogicalName != "account" &&
+                        request.OpportunityCustomerId.LogicalName != "contact")
+                    {
+                        throw new FaultException($"CustomerIdType for opportunity can either be an account or contact.");
+                    }
+
+                    var customer = db.GetDbRow(request.OpportunityCustomerId);
+
+                    if (customer == null)
+                    {
+                        throw new FaultException($"{customerMetadata.DisplayName} With Id = {request.OpportunityCustomerId.Id} Does Not Exist");
+                    }
+                }
+
+                var opportunity = new Entity
+                {
+                    Id = Guid.NewGuid(),
+                    LogicalName = "opportunity"
+                };
+
+                createdEntities.Add(opportunity.ToEntityReference());
+            }
+
+            var resp = new QualifyLeadResponse();
+            resp.Results["CreatedEntities"] = createdEntities;
```

Comment:
You also need to create the entities in the mockup database. Currently you just respond with almost the same data as the user has requested. Additionally, you also need to update the state of the Lead to Qualified.